rfcs/0136-stabilize-incrementally.md

```diff
@@ -0,0 +1,374 @@
+---
+feature: stabilize-incrementally
+start-date: 2022-09-15
+author: John Ericson
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: @thufschmitt @tomberek @infinisil
+shepherd-leader: @tomberek
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Ever since the closing of [RFC 49](https://github.com/NixOS/rfcs/pull/49), we've had the new CLI and Flakes marked as experimental, with no clear plan forward.
+
+With the goal of ending this current limbo and soothe longstanding tensions in the Nix community, this RFC does two things:
+
+1. Establish general principles about Nix's architecture and evolution in order to ensure we do not get in this situation again.
+
+   Notably we are allowed to make breaking changes to experimental features, which includes both the new CLI and Flakes, until they are stable.
+
+2. Establish an incremental plan adhering to the principles deciding on the order and priority in which to stabilize these features:
+
+   - First, the non-Flakes CLI will be stabilized, in phases corresponding to Nix's architecture.
+
+   - Afterwards, Flakes itself and its CLI components can be stabilized. The final design of Flakes will also require another RFC.
+
+# Problem statement
+[problem-statement]: #problem-statement
+
+For the past few years, we've accumulated a massive backlog of unstable features in the form of the new command-line interface and flakes.
+There is now a growing desire to stabilize those features, but as of yet no plan on exactly how to do so.
+The tension between that desire and a lack of a clear plan has loomed over our heads for a while; this RFC aims to dispel it by providing a concrete plan, an plan that hopefully will mitigate the lingering controversies and tensions around Flakes.
+
+## The situation so far
+
+There are a few facts that must be introduced to give context to this situation.
+We can't change these short of a time machine, so we must find a way to live with them instead.
+
+### Flakes are very popular
+
+As measured in the community polls, Nix has a lot of new users, dwarfing the number of long-time users.
+Flakes is very popular among these new users.
```

Comment:
Here's some **new** off-site documentation where people assume a general audience and do their best to write quite good tutorials which both take flakes as a starting point. I'm coming in from these and I imagine lots more people given how popular both these people are. Switching from flakes to non-flakes Nix for me at this point is jarring.

* https://mitchellh.com/writing/nix-with-dockerfiles
* https://fasterthanli.me/series/building-a-rust-service-with-nix/part-10